src/Simulation/Simulators.Tests/CoreTests.cs

```diff
@@ -324,5 +324,11 @@ public void CatchFail()
         [Fact]
         public void InternalCallables() =>
             OperationsTestHelper.RunWithMultipleSimulators(s => Circuits.InternalCallablesTest.Run(s).Wait());
+
+        [Fact]
+        public void CreateArrayWithNegativeSize() =>
+            Assert.ThrowsAny<Exception>(() =>
```

Comment:
nit: I usually like to have the specific exception type in here rather than the base class, but that's not a big deal.

Reply:
I wanted to use the specific exception type too, but bug #536 causes the wrong exception type to be thrown, and I didn't want to depend on the buggy behavior in this test case.

Reply:
Added a comment to the test about the exception type bug.